Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Extensions/Editors/ModulePackageEditor.cs

```diff
@@ -64,56 +64,9 @@ public bool SavePackageSettings(PackageSettingsDto packageSettings, out string e
-                    {
-                        var settingValue = packageSettings.EditorActions[settingName];
-
-                        switch (settingName.ToLowerInvariant())
-                        {
-                            case "foldername":
-                                desktopModule.FolderName = settingValue;
-                                break;
-                            case "category":
-                                desktopModule.Category = settingValue;
-                                break;
-                            case "businesscontroller":
-                                desktopModule.BusinessControllerClass = settingValue;
-                                break;
-                            case "dependencies":
-                                desktopModule.Dependencies = settingValue;
-                                break;
-                            case "hostpermissions":
-                                desktopModule.Permissions = settingValue;
-                                break;
-                            case "premiummodule":
-                                desktopModule.IsPremium = Convert.ToBoolean(settingValue);
-                                break;
-                            case "shareable":
-                                desktopModule.Shareable = (ModuleSharing)Convert.ToInt32(settingValue);
-                                break;
-                            case "assignportal":
-                                AssignPortals(desktopModule, JsonConvert.DeserializeObject<IList<ListItemDto>>(settingValue));
-                                break;
-                            case "unassignportal":
-                                UnassignPortals(desktopModule, JsonConvert.DeserializeObject<IList<ListItemDto>>(settingValue));
-                                break;
-                            case "savedefinition":
-                                var definition = JsonConvert.DeserializeObject<ModuleDefinitionDto>(settingValue);
-                                SaveModuleDefinition(definition);
-                                break;
-                            case "deletedefinition":
-                                DeleteModuleDefinition(Convert.ToInt32(settingValue));
-                                break;
-                            case "savemodulecontrol":
-                                var moduleControl = JsonConvert.DeserializeObject<ModuleControlDto>(settingValue);
-                                SaveModuleControl(moduleControl);
-                                break;
-                            case "deletemodulecontrol":
-                                DeleteModuleControl(Convert.ToInt32(settingValue));
-                                break;
-                        }
-                    }
+                {
+                    this.UpdateModuleProperties(desktopModule, packageSettings.Settings);

```

Comment:
Unless I'm missing something here in the process, why are we looping twice now through properties with settings and editoractions.  As settings was not used before?

Reply:
It was updating editoractions only for modules before this change.

Reply:
@mitchelsellers I think this looks right, there are two collections from the client which need contain actions/settings to process on the server.